### PR TITLE
progress reporting for Direct Image Migration

### PR DESCRIPTION
**Pull request by sseago**
This will need a mig-operator PR for the CRD changes.

### CODE REVIEW

**Comment by sseago** [comment]
Required mig-operator PR: https://github.com/konveyor/mig-operator/pull/526

File: pkg/apis/migration/v1alpha1/directimagemigration_types.go [hunk 1]
```diff
@@ -147,11 +154,57 @@ func (r *DirectImageMigration) HasErrors() bool {
...
+	progress = append(progress, r.getDISMProgress(r.Status.FailedISs, "Failed")...)
+	progress = append(progress, r.getDISMProgress(r.Status.DeletedISs, "Deleted")...)
+
+	return completed, reasons, progress
+}
+
+func (r *DirectImageMigration) getDISMProgress(items []*ImageStreamListItem, state string) []string {
+	progress := []string{}
+	for _, item := range items {
+		is := fmt.Sprintf("ImageStream %s (dism %s): %s ", path.Join(item.Namespace, item.Name), path.Join(item.DirectMigration.Namespace, item.DirectMigration.Name), state)
```
[the remark this answers is not in the document]
Comment (dymurray) [reply pkg/apis/migration/v1alpha1/directimagemigration_types.go:204]: This does leave a long string in the UI. Let's create a TODO issue in the UI repo to make this a column

**Review by dymurray — APPROVED** [review]
Worked for me looks great!

File: pkg/apis/migration/v1alpha1/directimagemigration_types.go [hunk 2]
```diff
@@ -35,19 +37,24 @@ type DirectImageMigrationSpec struct {
...
-	Namespace     string `json:"namespace,omitempty"`
-	Name          string `json:"name,omitempty"`
-	DestNamespace string `json:"destNamespace,omitempty"`
-	NotFound      bool   `json:"notFound,omitempty"`
+	*kapi.ObjectReference `json:",inline"`
+	DestNamespace         string                `json:"destNamespace,omitempty"`
+	NotFound              bool                  `json:"notFound,omitempty"`
+	DirectMigration       *kapi.ObjectReference `json:"directMigration,omitempty"`
+	Errors                []string              `json:"errors,omitempty"`
+	//Done                  bool                  `json:"done,omitempty"`
```
[the remark this answers is not in the document]
Comment (sseago) [reply pkg/apis/migration/v1alpha1/directimagemigration_types.go:57]: I forgot to remove it. The initial version of this struct had the "done" field but I never ended up using it. I commented it out to make sure removal didn't cause a compile error, but I forgot to return and remove it completely. I'll update the PR.

**Pull request merged**